### PR TITLE
calculate the reward

### PR DESCRIPTION
**Pull request by kuangpenghao**
calculate the reward

### CODE REVIEW

File: pkg/queries/user.go
Comment (Prince213) [inline pkg/queries/user.go:1]: Move these lines up to the invitation code check block.

File: pkg/models/user.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ type User struct {
 	IsAdmin          bool `gorm:"default:false"`
 	IsCommunityAdmin bool `gorm:"default:false"`
 	InvitationCode   string
+	Reward           int  `gorm:"default:0"`
```
Comment (Prince213) [inline pkg/models/user.go:32]: No need to set default value here, it's already the default.

```suggestion
	Reward           int
```

File: pkg/models/comment.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ type Comment struct {
 	CoverTitle          string
 	CoverContent        string
 	CoverReason         string
+	Reward			    int  `gorm:"default:0"`
```
Comment (Prince213) [inline pkg/models/comment.go:33]: No need to set default value here, it's already the default.

```suggestion
	Reward			    int
```

File: internal/controllers/comments/user.go [hunk 3]
```diff
@@ -117,6 +117,15 @@ func GenerateResponse(comments []models.Comment, uid uint, likeResult []CommentL
...
 			continue
 		}
+
+		user,err:=queries.GetUserByID(nil,v.User.ID)
+		if err==nil{
+			if user.InvCanReward == true{
+				user.Reward+=1
+				user.InvCanReward=false
+			}
+		}
```
Comment (Prince213) [inline internal/controllers/comments/user.go:127]: Always format your code before submitting.

File: pkg/models/user.go [hunk 4]
```diff
@@ -29,6 +29,8 @@ type User struct {
...
+	InvCanReward   	 bool  `gorm:"default:false"`
```
Comment (Prince213) [inline pkg/models/user.go:33]: Avoid bad abbreviation. Also its purpose is not obvious by looking at the name.

**Review by Prince213 — CHANGES_REQUESTED** [review]
Things to improve:
- [ ] Change pull request title
- [ ] Use proper commit messages.

File: pkg/queries/user.go [hunk 5]
```diff
@@ -145,6 +145,29 @@ func Register(db *gorm.DB, u *models.User, invitation_code string) error {
 		return errors.Wrap(err, errors.DatabaseError)
 	}
 
+
+
+	
+	//若该用户通过邀请码注册
+	if u.InvitationCode != ""{
```
Comment (AstatineAi) [inline pkg/queries/user.go:152]: `u.InvitationCode` is the invitation code generated in the current function for the registered user, not someone else's invitation code that the new user used to register. Use the `invitation_code` parameter of this function.

File: pkg/queries/user.go [hunk 6]
```diff
@@ -145,6 +145,29 @@ func Register(db *gorm.DB, u *models.User, invitation_code string) error {
...
+			return errors.Wrap(err, errors.DatabaseError)
+		}
+		if result.RowsAffected > 0 {
+			if user.IsActive {
+				return errors.New(errors.UserEmailDuplicated)
+			} else {
+				db.Delete(user)
+			}
+		}
+		userInvite.Reward += 1
```
Comment (AstatineAi) [inline pkg/queries/user.go:166]: `1` represents 0.01 yuan, please refer to the docs.

**Review by AstatineAi — CHANGES_REQUESTED** [review]

File: internal/controllers/comments/user.go
Comment (WAAutoMaton) [inline internal/controllers/comments/user.go:1]: 1. 计算 reward 的代码不应该放在 GenerateResponse 方法内（功能和方法名字不符）。我的建议是写在 `comments/Post` 方法内。
2. 对 User 示例修改后需要调用 `Save` 保存，否则修改不会写入数据库。
3. 获得 reward 的应该是邀请该用户的用户，而不是发表评论的用户自身
4. 整个修改过程应该用事务包起来，并且应该和发表评论作为同一个事务，建议是写在  `comments/post.go:106` 中 ：

```go
// 插入评论作为一个事务，若插入失败则回滚
err = db.Transaction(func(tx *gorm.DB) error {
	...
}
```

File: pkg/queries/user.go
Comment (WAAutoMaton) [inline pkg/queries/user.go:1]: 这部分代码更适合放在 register_active 的逻辑中

此外这部分也应该使用事务包起来

**Comment by Prince213** [comment]
Closing due to no activity.

**Pull request closed without merging**